### PR TITLE
POSIX socket: source addr selection

### PR DESCRIPTION
**Pull request by OlegHahm**
~~Depends on #4326~~

### CODE REVIEW

**Comment by cgundogan** [comment]
needs a rebase

**Comment by OlegHahm** [comment]
rebased

File: sys/include/net/conn.h [hunk 1]
```diff
@@ -38,11 +39,24 @@
...
 #endif
 
+/**
+ * @brief   Find the best matching source address for a given prefix
+ *
+ * @param[in] dst   Pointer to the IPv6 address to find a match for
+ *                  Must not be NULL
+ *
+ * @return NULL if no match address on any interface could be found
+ * @return pointer to an IPv6 configured on an interface with the best match to
```
Comment (cgundogan) [inline sys/include/net/conn.h]: pointer to an IPV6 **address** ...

Comment (OlegHahm) [inline sys/include/net/conn.h]: addressed

File: sys/include/net/conn.h [hunk 2]
```diff
@@ -38,11 +39,24 @@
...
 extern "C" {
 #endif
 
+/**
+ * @brief   Find the best matching source address for a given prefix
+ *
+ * @param[in] dst   Pointer to the IPv6 address to find a match for
+ *                  Must not be NULL
+ *
+ * @return NULL if no match address on any interface could be found
```
Comment (cgundogan) [inline sys/include/net/conn.h]: ~~match~~

Comment (OlegHahm) [inline sys/include/net/conn.h]: addressed

File: sys/posix/sockets/posix_sockets.c [hunk 3]
```diff
@@ -862,11 +864,15 @@ ssize_t sendto(int socket, const void *buffer, size_t length, int flags,
                                       sport, byteorder_ntohs(port));
             }
             else if (address != NULL) {
-                ipv6_addr_t local;
+                ipv6_addr_t unspec;
```
Comment (cgundogan) [inline sys/posix/sockets/posix_sockets.c]: you could use `ipv6_addr_t unspec = IPV6_ADDR_UNSPECIFIED;` here and remove the line below: `ipv6_addr_set_unspecified(&unspec);`

Comment (OlegHahm) [inline sys/posix/sockets/posix_sockets.c]: Do you think that makes any difference? If yes, a positive one? If yes, shouldn't we modify/remove this function then?

Comment (cgundogan) [inline sys/posix/sockets/posix_sockets.c]: my proposal adds another 12 bytes to ROM (iotlab-m3). outrages. leave as is

**Comment by cgundogan** [comment]
Looks sane. Please squash.

**Comment by OlegHahm** [comment]
squashed

**Comment by cgundogan** [comment]
travis agrees - GO

**Comment by OlegHahm** [comment]
Thx for the review.

**Pull request merged**